_posts/2016-02-07-coupling-in-distributed-systems.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+name: Coupling-in-distributed-systems
+title: 'Coupling in distributed systems'
+date: 2016-02-07 00:20:00 +00:00
+author: Felipe Fernández
+image:
+    src: /assets/img/custom/blog/coupling.png
+tags:
+- distributed systems
+- microservices
+- coupling
+
+---
+
+Coupling and cohesion are key quality indicators. We strive for systems highly cohesive and loosely coupled, but high doesn't mean pure. The same goes with functional programming, we aim for isolating and reducing side effects, but we need them unless we want a useless system. It's good to modularise our systems, so whenever those modules need to talk each other they'll effectively couple themselves. Our work is create cohesive modules so that coupling is minimised.
```

Comment:
A small side-note and not necessarily a suggestion to change, but your last sentence on this line makes it sound like cohesion and coupling are at opposite ends of the same spectrum, which they're not. It's quite possible to have a loosely coupled system with low cohesion, for example.